lib/capability-tool.js

```diff
@@ -0,0 +1,158 @@
+/**
+* Class for checking capability and resource support in a capability statement
+*
+*/
+class CapabilityTool {
+  /**
+  * Access capabilities
+  *
+  * @param {Object} capabilityStatement - capability statement FHIR resource
+  */
+  constructor(capabilityStatement) {
+    this.capabilityStatement = capabilityStatement;
+  }
+
+  /**
+  * Return capability support information based on the capability statement.
+  *
+  * @example
+  *
+  * const capabilities = new CapabilityTool(capabilityStatement);
+  *
+  * // Capability support
+  * const conditionalCreateSupport = capabilities.supportFor({
+  *   resourceType: 'Patient',
+  *   capabilityType: 'conditionalCreate'
+  * });
+  * console.log(patientReadSupport);
+  *
+  * // Capability support for specific code
+  * const patientReadSupport = capabilities.supportFor({
+  *   resourceType: 'Patient',
+  *   capabilityType: 'interaction',
+  *   where: { code: 'read' }
+  * });
+  * console.log(patientReadSupport);
+  *
+  * // Capability support for specific name
+  * const patientBirthDateSearchSupport = capabilities.supportFor({
+  *   resourceType: 'Patient',
+  *   capabilityType: 'searchParam',
+  *   where: { name: 'birthdate' }
+  * });
+  * console.log(patientBirthDateSearchSupport);
+  *
+  * @param {Object} params - The capability parameters.
+  * @param {String} params.resourceType - Resource type (e.g. 'Patient', 'Observation').
+  * @param {Object} params.capabilityType - Capability type (e.g. 'interaction', 'searchParam').
```

Comment:
I think we should put `[]` around all the optional params here, which should be all of them except `capabilityType`

Reply:
I see from the code it seems to require `resourceType`. Shouldn't we be able to check server-level capabilities, such as seeing if the system can support transactions:

`capabilities.supportFor({capabilityType: 'interaction', where: { code: 'transaction' }})`